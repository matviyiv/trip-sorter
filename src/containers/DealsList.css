.deals-list--item-heading {
  float: left;
}

.deals-list--item-price {
  float: right;
}
.deals-list {
  margin-right: 0;
  margin-bottom: 0;
  margin-left: 0;
  border-width: .2rem;
  position: relative;
  padding: 1rem;
  margin: 1rem -15px;
}
.deals-list .btn {
  float: right;
}
.green {
  color: green;
}
.red {
  color:red;
}