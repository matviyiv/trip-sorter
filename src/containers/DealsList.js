import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { getRealCost, durationToMin, durationToString } from '../utils';

export default class SearchForm extends Component {
  static propTypes = {
    triggerReset: PropTypes.func,
    tripRoute: PropTypes.array,
    deals: PropTypes.array,
    currency: PropTypes.string,
  };

  static defaultProps = {
    triggerReset: () => {},
    tripRoute: [],
    deals: [],
    currency: '',
  };

  onReset = () => {
    this.props.triggerReset();
  }

  renderRoutes() {
    const { tripRoute, deals, currency } = this.props;
    const dealsList = tripRoute.map((reference) => deals.find((deal) => deal.reference === reference));
    if (dealsList.length === 0) {
      return (<div>Sorry, no connection found</div>);
    }

    const routesList = [];
    let totalPrice = 0;
    let totalDuration = 0;

    dealsList.forEach((deal) => {
      const realCost = getRealCost(deal);
      routesList.push((
        <li key={deal.reference}>
          <div>{deal.departure} > {deal.arrival}</div>
          {deal.discount ?
            <div><s>{deal.cost}{currency}</s>{realCost}{currency}</div> :
            <div>{deal.cost}{currency}</div>
          }
          <div>{deal.transport} {deal.reference} for {deal.duration.h}h {deal.duration.m}</div>
        </li>
      ));
      totalPrice += realCost;
      totalDuration += durationToMin(deal.duration);
    });

    routesList.push((<li key="total">Total: {durationToString(totalDuration)} {totalPrice}{currency}</li>));

    return routesList;
  }

  render() {
    return (
      <div>
        <h2>DealsList</h2>
        <ul>
          {this.renderRoutes()}
        </ul>
        <button onClick={this.onReset}>Reset</button>
      </div>
    )
  }
}