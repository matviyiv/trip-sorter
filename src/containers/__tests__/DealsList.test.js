import React from 'react';
import { shallow } from 'enzyme';
import DealsList from '../DealsList';
import data from '../../../public/api/response.json';

describe('DealsList', () => {
  let deals;

  beforeEach(() => {
    deals = data.deals;
  });

  it('renders without crashing', () => {
    const wrapper = shallow(<DealsList />);
    const title = <h2>Route description</h2>;
    expect(wrapper.contains(title)).toEqual(true);
  });

  it('renders route description', () => {
    const tripRoute = [deals[0].reference];
    const wrapper = shallow(<DealsList deals={deals} tripRoute={tripRoute}/>);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('no route found', () => {
    const wrapper = shallow(<DealsList deals={deals} tripRoute={[]} />);
    const message = <div>Sorry, no connection found</div>;
    expect(wrapper.contains(message)).toEqual(true);
  })

});