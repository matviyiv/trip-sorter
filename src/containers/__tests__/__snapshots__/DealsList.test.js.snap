// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DealsList renders route description 1`] = `"<div><h2>Route description</h2><ul><li><div>London &gt; Amsterdam</div><div>160</div><div>train TLA0500 for 05h 00</div></li><li>Total: 5h0 160</li></ul><button>Reset</button></div>"`;
